dask_expr/tests/test_shuffle.py

```diff
@@ -372,6 +372,83 @@ def test_sort_head_nlargest(df):
     b = df.nsmallest(10, columns=["x"]).expr
     assert a.optimize()._name == b.optimize()._name
 
+    a = df.sort_values("x", ascending=[False]).head(10, compute=False).expr
+    b = df.nlargest(10, columns=["x"]).expr
+    assert a.optimize()._name == b.optimize()._name
+
+    a = df.sort_values("x", ascending=[True]).head(10, compute=False).expr
+    b = df.nsmallest(10, columns=["x"]).expr
+    assert a.optimize()._name == b.optimize()._name
+
+    a = df.sort_values(["x"], ascending=[False]).head(10, compute=False).expr
+    b = df.nlargest(10, columns=["x"]).expr
+    assert a.optimize()._name == b.optimize()._name
+
+    a = df.sort_values(["x"], ascending=[True]).head(10, compute=False).expr
+    b = df.nsmallest(10, columns=["x"]).expr
+    assert a.optimize()._name == b.optimize()._name
+
+    a = df.sort_values(["x", "y"], ascending=[False]).head(10, compute=False).expr
+    b = df.nlargest(10, columns=["x", "y"]).expr
+    assert a.optimize()._name == b.optimize()._name
+
+    a = df.sort_values(["x", "y"], ascending=[True]).head(10, compute=False).expr
+    b = df.nsmallest(10, columns=["x", "y"]).expr
+    assert a.optimize()._name == b.optimize()._name
+
+    with pytest.raises(
+        NotImplementedError,
+        match="Dask currently only supports a single boolean for ascending",
+    ):
+        a = (
+            df.sort_values(["x", "y"], ascending=[False, True])
+            .head(10, compute=False)
+            .expr
+        )
+
+
+def test_sort_single_partition_head_nlargest():
+    pdf = pd.DataFrame({"x": np.random.random(100), "y": np.random.random(100)})
+    df = from_pandas(pdf, npartitions=1)
+
+    result = df.sort_values("x", ascending=False).head(10)
+    expected = pdf.sort_values("x", ascending=False).head(10)
+    assert_eq(result, expected, sort_results=False)
+
+    result = df.sort_values("x", ascending=True).head(10)
+    expected = pdf.sort_values("x", ascending=True).head(10)
+    assert_eq(result, expected, sort_results=False)
+
+    result = df.sort_values("x", ascending=[False]).head(10)
+    expected = pdf.sort_values("x", ascending=[False]).head(10)
+    assert_eq(result, expected, sort_results=False)
+
+    result = df.sort_values("x", ascending=[True]).head(10)
+    expected = pdf.sort_values("x", ascending=[True]).head(10)
+    assert_eq(result, expected, sort_results=False)
+
+    result = df.sort_values(["x"], ascending=[True]).head(10)
+    expected = pdf.sort_values(["x"], ascending=[True]).head(10)
+    assert_eq(result, expected, sort_results=False)
+
+    result = df.sort_values(["x"], ascending=[False]).head(10)
+    expected = pdf.sort_values(["x"], ascending=[False]).head(10)
```

Comment:
This case also fails in the current implementation, which is why https://github.com/coiled/benchmarks/pull/1290/files#diff-ed9173e9cde3ccc498e25c8485d7c2060041c6b7cb0228f26e1c16603a3f3b10L478 works as a fix.